witness/src/Dscp/Witness/Web/Logic.hs

```diff
@@ -246,9 +247,17 @@ checkFairCVPDF
     => Pdf.PDFBody -> m FairCVAndCheckResult
 checkFairCVPDF pdf = do
     let maybeFairCV = do
-            fairCVencoded <- Pdf.project (Pdf.MaxSearchLength (Just 2048)) pdf
-            Aeson.decodeStrict fairCVencoded
+            (fairCVencoded, source) <- Pdf.unInject (Pdf.MaxSearchLength (Just 2048)) pdf
+            unenc <- Aeson.decodeStrict fairCVencoded
+            return (unenc, source)
+
+    (fairCV, Pdf.PDFBody source) <- maybe (throwM InvalidFormat) pure maybeFairCV
+
+    unless
+        (  fairCV^?to fcCV.each.each.to toList.each.ptSignedSubmission.ssSubmission.sContentsHash
+        == Just (coerce $ hash source)
+        ) $ do
+            throwM InvalidFormat
```

Comment:
_That's some **thicc** traversals over here_

Hoverer, as I said before, you should only have the PDF hash in one of those transactions. Also, `InvalidFormat` is not the thing we want to see there - this is a legit case of correctly formatted yet invalid FairCV, we need to return false `FairCVCheckResult` here.